kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -270,6 +277,117 @@ def set_phase_parameters(
         phase = {k: v for k, v in inputs.items() if v is not None}
         _update_phase_info(self.metadata, phase, number)
 
+    def get_patterns(
+        self,
+        rotations: Rotation,
+        detector: EBSDDetector,
+        energy: int,
+        chunk_size: int,
+        dtype_out=np.float32,
+    ) -> LazyEBSD:
+        """
+        Creates a catalogue of EBSD patterns given a set of rotations and
+        a detector model from the master pattern.
+
+        Parameters
+        ----------
+        rotations : Rotation
+            Set of rotations to get patterns from.
+        detector : EBSDDetector
+            EBSDDetector object describing the detector geometry.
+        energy : int
+            The wanted energy in the master pattern.
+        chunk_size : int
+            The amount of rotations the dask arrays should work on per chunk.
+        dtype_out : numpy.dtype
+            Data type of the returned patterns.
+
+        Returns
+        ----------
+        LazyEBSD object containing the simulated EBSD patterns.
+        """
+
+        if (
+            self.metadata.Simulation.EBSD_master_pattern.Master_pattern.projection
+            != "lambert"
+        ):
+            raise NotImplementedError(
+                "Method only supports master patterns in lambert projection!"
+            )
+
+        dc = _get_direction_cosines(detector)
```

Comment:
I think this calculation and the dask array initialization of the rotation data below should be done after all checks for valid parameters etc. are done, like the centrosymmetry check discussed and the axes checks below, what do you think?

Reply:
Sounds reasonable!